assets/js/base/components/product-list/style.scss

```diff
@@ -222,6 +199,70 @@
 	}
 }
 
+.editor-styles-wrapper .wc-block-grid__products .wc-block-grid__product .wc-block-grid__product-onsale,
+.wc-block-grid__product-onsale {
+	border: 1px solid #43454b;
+	color: #43454b;
+	background: #fff;
+	padding: 0.202em 0.6180469716em;
+	font-size: 0.875rem;
+	text-align: center;
+	text-transform: uppercase;
+	font-weight: 600;
+	display: inline-block;
+	width: auto;
+	border-radius: 3px;
+	z-index: 9;
+	position: relative;
+	margin: $gap-smaller auto;
+
+	&.wc-block-grid__product-onsale--alignleft {
+		float: left;
+		margin: 0 auto 0 0;
+	}
+	&.wc-block-grid__product-onsale--alignright {
+		float: right;
+		margin: 0 0 0 auto;
+	}
+}
+.editor-styles-wrapper .wc-block-grid__products .wc-block-grid__product .wc-block-grid__product-image,
+.wc-block-grid__product-image {
+	.wc-block-grid__product-onsale {
+		&.wc-block-grid__product-onsale--alignleft {
+			position: absolute;
+			left: $gap-smaller/2;
+			top: $gap-smaller/2;
+			right: auto;
+			margin: 0;
+		}
+		&.wc-block-grid__product-onsale--aligncenter {
+			position: absolute;
+			top: $gap-smaller/2;
+			left: auto;
+			right: auto;
```

Comment:
Actually that's not center exactly in the middle. Adding these declarations (and removing `right: auto`) should do the trick:
```CSS
left: 50%;
transform: translateX(-50%);
```